### ISSUE TITLE
azurerm_postgresql_active_directory_administrator: "string is required" when upgrading from AzureRM 3.14 to 3.15

### ISSUE THREAD

**Issue by 9numbernine9** [hashicorp/terraform-provider-azurerm#17727]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.5

### AzureRM Provider Version

3.15

### Affected Resource(s)/Data Source(s)

azurerm_postgresql_active_directory_administrator

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "3.15.0"
    }
  }
}

provider "azurerm" {
  features {}
}

data "azurerm_client_config" "current" {}

resource "azurerm_resource_group" "example" {
  name     = "azurerm-debug"
  location = "canadaeast"
}

resource "azurerm_postgresql_server" "example" {
  name                         = "azurerm-debug-psqlserver"
  resource_group_name          = azurerm_resource_group.example.name
  location                     = azurerm_resource_group.example.location
  version                      = "11"
  administrator_login          = "administratorexample"
  administrator_login_password = "thisisaterriblepassworddontuseit12345!"
  sku_name                     = "GP_Gen5_2"
  ssl_enforcement_enabled      = true
}

resource "azurerm_postgresql_active_directory_administrator" "example" {
  server_name         = azurerm_postgresql_server.example.name
  resource_group_name = azurerm_resource_group.example.name

  login               = "sqladmin"
  tenant_id           = data.azurerm_client_config.current.tenant_id
  object_id           = data.azurerm_client_config.current.object_id
}
```

### Debug Output/Panic Output

```shell
https://gist.github.com/9numbernine9/bd59bd2fb995cb5d8c1f85f01dee16b6
```

### Expected Behaviour

Terraform will `plan` and/or `apply` successfully.

### Actual Behaviour

The following error is generated:
```
╷
│ Error: string is required
│ 
│   with azurerm_postgresql_active_directory_administrator.example,
│   on main.tf line 32, in resource "azurerm_postgresql_active_directory_administrator" "example":
│   32: resource "azurerm_postgresql_active_directory_administrator" "example" {
│ 
╵
```

### Steps to Reproduce

1. Change the Terraform configuration in this issue to use AzureRM `3.14`.
2. `terraform init -upgrade` 
3. `terraform apply` - this should work fine.
4. Switch to AzureRM `3.15`
5. `terraform init -upgrade`
6. `terraform plan` - Error message received.

### Important Factoids

This only occurs when upgrading a resource that was previously created with AzureRM `3.14` (possibly earlier versions as well). This error will _not_ occur if the resource is created from scratch using AzureRM `3.15`.

**Comment by md2k** [comment]
Have same problem. Snip from Trace:    

```
2022-07-22T17:03:27.914+0200 [ERROR] plugin.terraform-provider-azurerm_v3.15.0_x5: Response contains error diagnostic: diagnostic_severity=ERROR diagnostic_summary="string is required" tf_provider_addr=provider @caller=github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/internal/diag/diagnostics.go:56 tf_resource_type=azurerm_postgresql_active_directory_administrator diagnostic_attribute=AttributeName("id") tf_proto_version=5.2 tf_req_id=XXX-XXXX-XXXX-XXXXX tf_rpc=UpgradeResourceState @module=sdk.proto diagnostic_detail= timestamp=2022-07-22T17:03:27.914+0200
```

**Comment by favoretti** [comment]
Hi there and thank you for reporting this issue. Seems related to:
```
2022-07-22T09:21:45.124-0400 [DEBUG] provider.terraform-provider-azurerm_v3.15.0_x5: Updating ID from "/subscriptions/650efc9f-033a-46a5-8e1a-294c9be627d2/resourceGroups/azurerm-debug/providers/Microsoft.DBforPostgreSQL/servers/azurerm-debug-psqlserver/administrators/activeDirectory" to "Server (Subscription: \"650efc9f-033a-46a5-8e1a-294c9be627d2\"\nResource Group Name: \"azurerm-debug\"\nServer Name: \"azurerm-debug-psqlserver\")": timestamp=2022-07-22T09:21:45.123-0400
```

I'll try to have a look today.

**Comment by favoretti** [comment]
Got a potential fix, testing.

**Comment by mbfrahry** [comment]
Hey @9numbernine9! Sorry for the regression here. We got a fix in with #17732 and just pushed a release out to 3.15.1. Let me know if that fixes up this issue for you!

**Comment by 9numbernine9** [comment]
@mbfrahry The new version does appear to fix the issue! Thanks for the quick fix! :+1:
